### ISSUE TITLE
JupyterHub Helm chart repository health checkup

### ISSUE THREAD

**Issue by consideRatio** [pangeo-data/jupyter-earth#12]
I made a push to get the JupyterHub Helm chart repository in better shape. It had about 330 open issues and several PRs that could not be merged as they were. During this push, I've now closed ~250 issues, created/contributed to than 20 PRs, and given proper feedback and taken action on all stale PRs where i often augmented them to be mergeable.

Most of [the PRs I created](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pulls?q=is%3Apr+author%3Aconsideratio+created%3A2020-10-02..2020-10-08) were meant to close issues reported, and the kind of changes include: bug fixes, new configuration options, security enhancements, guide documentation updates.

There is still some work to get some PRs to get merged following review feedback, and I'd also want to process [an Issue triage report](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1810) I created in order to try summarize as I gained a high level view of all the issues reported. The most common issue reported related to the Helm chart configuration of chosen JupyterHub Authenticator. The problem is that we have created a custom syntax to configure the authenticators through the Helm chart configuration, and have had to manually maintain a mapping to the actual configuration of the various Authenticators.

## Work I'd like to follow this up with
- [ ] Process the [issue triage report](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1810) to help other contributors take action to solve issues.
- [ ] Find a sustainable solution for the most common issue regarding Authenticator configuration.

**Comment by choldgraf** [comment]
Nice - really great work Erik. Cleaning up issues is a huge boost to signal-to-noise and will hopefully make it easier to continue maintaining in the future. Many many thanks

**Comment by consideRatio** [comment]
Z2JH 0.10.0-beta.1 is now released, and I wanted to give an indication of the push Jupyter Earth have enabled me to do in order to contribute towards this release and in general vitalize the Z2JH repo a bit.

![image](https://user-images.githubusercontent.com/3837114/97034940-eef75980-1565-11eb-8c06-2de98cc49725.png)

**Comment by choldgraf** [comment]
Amazing - @consideRatio I wonder if it's a good moment to produce a short update blog post or something like this? I would be happy to help you write this if you think there is enough content to talk about.

**Comment by consideRatio** [comment]
> Amazing - @consideRatio I wonder if it's a good moment to produce a short update blog post or something like this? I would be happy to help you write this if you think there is enough content to talk about.

Thanks for the suggestion and encouragement! There is content to talk about, but the question is what narrative to choose that creates an interesting narrative that has a start and finish and includes the content?

## Tech depth post
It think I would feel most at ease writing something quite tech oriented, such pieces could become relevant references for z2jh or similarly, for example something like...
- About NetworkPolicies which I've recently spent a lot of time on
- About PodSecurityPolicies and securityContext for pods / containers

## Tech overview post
- An overview of the z2jh helm chart and how various pieces are meant to work together

## Soft focus post
- About my experience of working on things surrounding the z2jh Helm chart.

---

I'm not sure on the incentives to optimize for, but I'd be most motivated to write a tech overview or in depth tech post as it could help admins of z2jh / binderhub etc to better understand how to use the helm charts. I'd love your input and help with directions to get started on something @choldgraf!

**Comment by consideRatio** [comment]
I did a smaller push to update issue triage KubeSpawner as well, here are some results of that: https://github.com/jupyterhub/kubespawner/issues/456.

**Comment by consideRatio** [comment]
@choldgraf I discussed writing a blog post in the bi-weekly Jupyter-Earth meeting! With feedback from @lheagy I've concluded to try write a Blog post about the value outcomes from core maintenance work like this. I'm personally very pleased to have been able to spend time doing it and consider it to have great value, like soap - boring and important.

**Comment by choldgraf** [comment]
yes that sounds great! @yuvipanda and I were just discussing this today as well from the perspective of 2i2c. We should make sure to highlight all of the work that the team is doing for the broader open source community, not just building new features, and not just building things for ourselves, but for others as well.

I am happy to help edit/review/etc any blog posts!

**Comment by consideRatio** [comment]
## Update of core maintenance work
- Z2JH/JupyterHub releases (https://discourse.jupyter.org/t/jupyterhub-helm-chart-0-10-0-released-0-10-1-also-patch-release/6543)
- TravisCI migration to GitHub actions (https://github.com/jupyterhub/team-compass/issues/353)
- more to list...
